tutorial/go/fuzz/fuzz.go

```diff
@@ -0,0 +1,144 @@
+package thriftfuzz
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"context"
+	"fmt"
+	"shared"
+	"strconv"
+	"tutorial"
+
+	"github.com/apache/thrift/lib/go/thrift"
+)
+
+func fuzzChooseProtocol(d byte, t thrift.TTransport) thrift.TProtocol {
+	switch d {
+	case 0:
+		return thrift.NewTCompactProtocolFactory().GetProtocol(t)
+	}
+	return thrift.NewTBinaryProtocolFactoryDefault().GetProtocol(t)
```

Comment:
nit: those factory constructors are deprecated, please use `NewTCompactProtocolFactoryConf(nil)` and `NewTBinaryProtocolFactoryConf(nil)` instead.

Reply:
ok